lib/engine/game/g_steam_over_holland/game.rb

```diff
@@ -356,44 +361,48 @@ def non_floated_corporations
         end
 
         def check_distance(route, visits, train = nil)
-          super
-
           raise GameError, 'Route cannot begin/end in a town' if visits.first.town? || visits.last.town?
+
+          super
         end
 
         def revenue_for(route, stops)
-          super
+          revenue = super
+
+          abilities = abilities(route.corporation, :hex_bonus)
+          return revenue unless abilities
```

Comment:
I'd missed this last time, sorry.
```suggestion
          abilities = Array(abilities(route.corporation, :hex_bonus))
          return revenue if abilities.empty?
```
`Game.abilities` is, IMHO, a total pain. It can either return nil, a single ability object, or an array of ability objects. My approach is, unless you can be absolutely certain about what type will be returned, is to wrap the call inside `Array(…)`. That way you are guaranteed to get an array under all circumstances (`Array(nil) == []`, `Array(Object) = [Object]` and `Array([1, 2, 3]) = [1, 2, 3]`).

The code as currently written will work if zero or 2+ abilities are found. It will fail if there's a single matching ability, as `flat_map` isn't a method implemented by `Ability`.

Reply:
Ha, good point. Thanks. Fixed.

Reply:
done